pkg/mimir/mimir.go

```diff
@@ -530,12 +530,13 @@ func (t *Mimir) readyHandler(sm *services.Manager) http.HandlerFunc {
 			msg := bytes.Buffer{}
 			msg.WriteString("Some services are not Running:\n")
 
-			byState := sm.ServicesByState()
-			for st, ls := range byState {
-				msg.WriteString(fmt.Sprintf("%v: %d\n", st, len(ls)))
+			for name, s := range t.ServiceMap {
+				msg.WriteString(fmt.Sprintf("%s: %s\n", name, s.State()))
 			}
 
-			http.Error(w, msg.String(), http.StatusServiceUnavailable)
+			strMsg := msg.String()
+			level.Debug(util_log.Logger).Log(strMsg)
```

Comment:
There is no need to debug-log this.

Reply:
My original thinking was that this should make it easier to see why components aren't ready - no need to port-forward to pods and curl endpoints - it will be in the logs.

Do you think that would be useful?

Reply:
We already have logs for each starting/started module, which can be used to find out which modules aren't ready.

If we change this log line to only log non-ready modules, I think it would be more useful than logging ALL modules.

Reply:
> If we change this log line to only log non-ready modules, I think it would be more useful than logging ALL modules.

agreed, pushed a change in https://github.com/grafana/mimir/pull/2055/commits/d214659ea0087827d7a905de0b4838927514de4c